File: problems/elevator/solutions/elevator.py
# Python 2
import sys

capacity, people = map(int, raw_input().split())
floors = [int(line) for line in sys.stdin]

above = sorted((floor for floor in floors if floor > 0), reverse=True)
below = sorted((-floor for floor in floors if floor < 0), reverse=True)
one_way = sum(above[i] for i in xrange(0, len(above), capacity)) \
    + sum(below[i] for i in xrange(0, len(below), capacity))
distance = 2 * one_way - max(next(iter(above), 0), next(iter(below), 0))

hour = 9 + distance / 3 / 60 % 24
minute = distance / 3 % 60
second = distance % 3 * 20

print '{:02}:{:02}:{:02} {}'.format((hour - 1) % 12 + 1, minute, second, 'AM' if hour < 12 else 'PM')

File: problems/elevator/solutions/elevator.scala
import java.util.Scanner
import scala.util.control.Exception._

object Main {
  def main(args: Array[String]) {
    val input = new Scanner(System.in)
    val capacity = input.nextInt()
    val people = input.nextInt()
    val floors = Iterator.continually(input.nextInt()).take(people).toSeq
    val oneWay =
      floors.filter(_ > 0).sorted.reverse.grouped(capacity).map(_.head).sum +
      floors.filter(_ < 0).sorted.map(-_).grouped(capacity).map(_.head).sum
    val distance = 2 * oneWay - (try floors.map(math.abs).max catch { case e: UnsupportedOperationException => 0 })

    val hour = 9 + distance / 3 / 60 % 24
    val minute = distance / 3 % 60
    val second = distance % 3 * 20

    println(f"${(hour - 1) % 12 + 1}%02d:$minute%02d:$second%02d ${if (hour < 12) "AM" else "PM"}")
  }
}
